content/blog/pulumi-convert-terraform-improvements/index.md

```diff
@@ -0,0 +1,352 @@
+  organization = var.planetscale_org
+}
+
+# Generate a password for connecting to the DB
+resource "planetscale_password" "db_password" {
+  organization = "brandonpollack23"
+  name         = "terraform-generated"
+  database     = planetscale_database.db.name
+  branch       = planetscale_database.db.default_branch
+}
+
+# Spin up a small GCE VM
+resource "google_compute_instance" "vm" {
+  name         = "demo-vm"
+  machine_type = "e2-micro"
+  zone         = var.gcp_zone
+
+  # Minimal Debian boot disk
+  boot_disk {
+    initialize_params {
+      image = "debian-cloud/debian-11"
+    }
+  }
+
+  # Use the default VPC with a public IP
+  network_interface {
+    network = "default"
+    access_config {}
+  }
+
+  # Pass PlanetScale connection info as metadata
+  metadata = {
+    DB_HOST = planetscale_database.db.url
+    DB_USER = "root"
+    DB_PASS = planetscale_password.db_password.plaintext
+    DB_NAME = planetscale_database.db.name
+  }
+}
+```
+
+{{% notes type="warning" %}}
+**Never** store secrets or keys in plain text in your code or commited
+configuration files. I only have a db_password stored here in plain text for
+the sake of readability.
+
+Consider using [Pulumi ESC](https://www.pulumi.com/docs/esc/) to store these types of
+secrets (and any other configuration!) and access them directly from your Pulumi programs.
+{{% /notes %}}
+
+This alone works if you run `terraform plan`, however without *real*
+credentials nothing will really deploy, of course! You can set these in tfvars,
+environmental variables, command line, etc.  
+
+Even if you use a tfvars file, converting this is not currently supported, so
+the configuration will need to be moved to stack files in your
+new Pulumi project (eg `Pulumi.dev.yaml`, `Pulumi.prod.yaml`, etc).
+
+You can convert this into a new project with the following command:
+
+{{% chooser language "typescript,python,go" %}}
+
+{{% choosable language typescript %}}
+
+```shell
+pulumi convert --from terraform --language typescript --out ts
+```
+
+I have opted to output it to a different directory to preserve the contents of my Terraform project.
+
+### Project structure
+
+If we navigate to the `ts` directory, we can see a few things:
+
+- index.ts, our equivalent to main.tf for typescript projects
+- sdks directory, where the generated bridged provider is output
+- other node project artifacts such as package.json, node_modules, tsconfig.json, etc.
+
+{{% /choosable %}}
+
+{{% choosable language python %}}
+
+```shell
+pulumi convert --from terraform --language python --out python
+```
+
+I have opted to output it to a different directory to preserve the contents of my Terraform project.
+
+### Project structure
+
+If we navigate to the `python` directory, we can see a few things:
+
+- \_\_main\_\_.py, our equivalent to main.tf for python projects
+- sdks directory, where the generated bridged provider is output
+- other python project artifacts (eg requirments.txt, pyproject.toml, etc)
```

Comment:
"python" -> "Python" (capitalize proper nouns)

Reply:
"(eg ... )" -> "such as ..." (consistent language with previous example)

Reply:
"requirments.txt, pyproject.toml" -> "`requirements.txt`, `pyproject.toml`" (use code formatting for filenames, also "requirements" is misspelled)